curp/src/client.rs

```diff
@@ -323,12 +347,22 @@ where
                 .get_connect(leader_id)
                 .unwrap_or_else(|| unreachable!("leader {leader_id} not found"))
                 .wait_synced(
-                    WaitSyncedRequest::new(cmd.id()),
+                    WaitSyncedRequest::new(cmd.id(), self.cluster_version()),
                     *self.config.wait_synced_timeout(),
                 )
                 .await
             {
                 Ok(resp) => resp.into_inner(),
+                Err(RpcError::StatusError(status)) => {
+                    if is_wrong_cluster_version(&status) {
+                        return Err(CommandProposeError::Propose(
+                            ProposeError::WrongClusterVersion,
+                        ));
```

Comment:
Why should we return immediately if the cluster version is wrong? I think this early return may cause a long time of unavailability if a server is lag behind.

Reply:
Why it may cause a long time of unavailability? After returning error here, the outer propose will pull the cluster's configuration and retry the entire propose process

Reply:
I mean even if only one server is lagged behind, this method will return error, but the majority of the cluster may be actually available right now.

Reply:
The client may be using any cluster version of the configuration, so it cannot tell if most nodes are using the new configuration. And there are no correctness issues with the current implementation, it just has a small probability of slowing down requests. I think the current implementation can be used before there is a safe and fast way